src/main/java/org/jspecify/annotations/NullMarked.java

```diff
@@ -41,30 +41,27 @@
  * supporting analysis tools will be tracking the changes on varying schedules. Releasing a library
  * using these annotations in its API is <b>strongly discouraged</b> at this time.
  *
- * <h2>Effects of being null-marked</h2>
+ * <h2 id="effects">Effects of being null-marked</h2>
  *
  * <p>Within null-marked code, as a <i>general</i> rule, a type usage is considered non-null (to
  * exclude {@code null} as a value) unless explicitly annotated as {@link Nullable}. However, there
  * are several special cases to address.
  *
- * <h3 id="special-cases">Special cases</h3>
+ * <h3 id="effects-special-cases">Special cases</h3>
  *
  * <p>Within null-marked code:
  *
  * <ul>
- *   <li>Being null-marked has no consequence for any type usage where {@code @Nullable} and
- *       {@code @NonNull} are <a href="Nullable.html#applicability"><b>not applicable</b></a>, such
- *       as the root type of a local variable declaration.
  *   <li>An <b>unbounded wildcard</b> (the {@code ?} in a type like {@code List<?>}, not followed by
- *       {@code extends} or {@code super}) represents <i>any</i> type within the type parameter's
- *       bounds with no further restriction. A <b>lower-bounded wildcard</b> ({@code List<? super
- *       String>}) similarly has no <i>upper</i> bound of its own. Therefore, since <i>nullable</i>
- *       {@code Object} is the new "<a href="Nullable.html#subtypes">top type</a>", either list
- *       might have null elements, even though the word {@code @Nullable} is nowhere in sight. On
- *       the other hand, in {@code List<? extends Object>} a bound <i>is</i> provided, and as usual,
- *       being unannotated it is considered non-null. This means that while {@code List<?>} and
- *       {@code List<? extends Object>} have always been identical as base types, they are no longer
- *       identical as <a href="Nullable.html#augmented-types">augmented types</a>. (<a
+ *       {@code extends} or {@code super}) will represent <i>any</i> type within the type
+ *       parameter's bounds, with no further restriction. A <b>lower-bounded wildcard</b> ({@code
```

Comment:
This is a fairly confusing area (I know I was confused by it), so I think it might be worth spelling it out in even more detail. You might add text like this, for example:

> ...with no further restriction. So if {@code List} is declared as {@code interface List<E extends @Nullable Object>}
> then {@code List<?>} is equivalent to {@code List<? extends @Nullable Object>}.
> If {@code List} is {@code interface List<E>} or {@code List<E extends Object>}
> then {@code List<?>} is equivalent to {@code List<? extends Object>}.

Reply:
Well, this prompted me to attempt a rewording, see what you think.